clash-vexriscv/src/VexRiscv.hs

```diff
@@ -111,13 +125,21 @@ vexRiscv input =
 
   where
     (unbundle -> (timerInterrupt, externalInterrupt, softwareInterrupt, iBusS2M, dBusS2M))
-      = (<$> input) $ \(Input a b c d e) -> (a, b, c, d, e)
+    -- A hack that enables us to both synthesize and simulate vexRisc in the REPL
```

Comment:
It's not only about the REPL though :stuck_out_tongue: Maybe this can be worded differently because the REPL isn't really relevant? With this the same code can be executed as Haskell / "simulated" or generate synthesisable HDL